src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java

```diff
@@ -109,21 +171,56 @@ private static ArgumentMultimap extractArguments(String argsString, List<PrefixP
         return argMultimap;
     }
 
-    /**
-     * Returns the trimmed value of the argument in the arguments string specified by {@code currentPrefixPosition}.
-     * The end position of the value is determined by {@code nextPrefixPosition}.
-     */
     private static String extractArgumentValue(String argsString,
                                         PrefixPosition currentPrefixPosition,
                                         PrefixPosition nextPrefixPosition) {
         Prefix prefix = currentPrefixPosition.getPrefix();
 
-        int valueStartPos = currentPrefixPosition.getStartPosition() + prefix.getPrefix().length();
+        int valueStartPos = currentPrefixPosition.getStartPosition() + prefix.getLength();
+        String value = argsString.substring(valueStartPos, nextPrefixPosition.getStartPosition());
+
+        return value.trim();
+    }
+
+    private static ArgumentMultimap extractFilterArguments(String argsString, List<PrefixPosition> prefixPositions) {
+
+        prefixPositions.sort((prefix1, prefix2) -> prefix1.getStartPosition() - prefix2.getStartPosition());
+
+        PrefixPosition preambleMarker = new PrefixPosition(new Prefix(""), 0);
+        prefixPositions.add(0, preambleMarker);
+
+        PrefixPosition endPositionMarker = new PrefixPosition(new Prefix(""), argsString.length());
+        prefixPositions.add(endPositionMarker);
+
+        ArgumentMultimap argMultimap = new ArgumentMultimap();
+        for (int i = 0; i < prefixPositions.size() - 1; i++) {
+            Prefix argPrefix = prefixPositions.get(i).getPrefix();
+            String argValue = extractFilterArguments(argsString, prefixPositions.get(i), prefixPositions.get(i + 1));
+            argMultimap.put(argPrefix, argValue);
+        }
+
+        return argMultimap;
+    }
+
+    private static String extractFilterArguments(String argsString,
```

Comment:
Same as previous comment, a javadoc would be better suited as these are non-trivial methods

Reply:
Done